src/Plugin/Field/FieldFormatter/StrawberryPdfFormatter.php

```diff
@@ -0,0 +1,303 @@
+                // We assume here file could not be accessible publicly
+                $route_parameters = [
+                  'node' => $nodeid,
+                  'uuid' => $file->uuid(),
+                  'format' => 'default.'. pathinfo($file->getFilename(), PATHINFO_EXTENSION)
+                ];
+                $publicurl = Url::fromRoute('format_strawberryfield.iiifbinary', $route_parameters);
+                $uniqueid =
+                  'pdf-' . $items->getName(
+                  ) . '-' . $nodeuuid . '-' . $delta . '-document' . $i;
+
+                //@TODO make a select component that is ajax driven.
+                // If we have more than a single Document, simply rebuild and reload the given $delta instead of rendering
+                // Multiple deltas as i do here.
+                $elements[$delta]['controller' . $i] = [
+                  '#theme' => 'format_strawberryfield_pdfs',
+                  '#item' =>  [
+                    'id' =>  'document_' . $uniqueid,
+                  ]
+                ];
+                $elements[$delta]['pdf' . $i] = [
+                  '#type' => 'html_tag',
+                  '#tag' => 'canvas',
+                  '#attributes' => [
+                      'class' => ['field-pdf-canvas','strawberry-document-item'],
+                      'id' => 'document_' . $uniqueid,
+                      'style' => "max-width:{$max_width}px;height:{$max_height}",
+                      'data-iiif-document' =>  $publicurl->toString(),
+                      'data-iiif-initialpage' => $initial_page,
+                      'data-iiif-document-width' => $max_width,
+                      'data-iiif-document-height' => $max_height,
+                      'data-iiif-pages' => $number_pages,
+                    ],
+                   '#alt' => $this->t(
+                      'PDF @name for  @label',
+                      [
+                        '@label' => $items->getEntity()->label(),
+                        '@name' => $file->getFilename()
+                      ]),
+                  ];
+
+                  $elements[$delta]['pdf'.$i]['#attached']['drupalSettings']['format_strawberryfield']['pdf']['innode'][$uniqueid] = $nodeuuid;
+                  $elements[$delta]['#attached']['library'][] = 'format_strawberryfield/pdfs_strawberry';
+
+              }
+              else {
+                // @TODO Deal with no access here
+                // Should we put a thumb? Just hide?
+                // @TODO we can bring a plugin here and there that deals with
+                $elements[$delta]['pdf'.$i] = [
+                  '#markup' => '<i class="fas fa-times-circle"></i>',
+                  '#prefix' => '<span>',
+                  '#suffix' => '</span>',
+                ];
+              }
+            } elseif (isset($mediaitem['url'])) {
+              // TODO. We can serve non mananged by US PDFS directly here
+              // We would just have less data. But that is all
```

Comment:
Comment is deceiving. If the remote URL has no CORS enabled, then via JS we have no chance. Lets keep it this way